src/EnergyPlus/ZoneDehumidifier.cc

```diff
@@ -151,22 +151,23 @@ namespace ZoneDehumidifier {
             CompIndex = ZoneDehumidNum;
         } else {
             ZoneDehumidNum = CompIndex;
-            if (ZoneDehumidNum > state.dataZoneDehumidifier->NumDehumidifiers || ZoneDehumidNum < 1) {
+            int NumDehumidifiers = (int)state.dataZoneDehumidifier->ZoneDehumid.size();
+            if (ZoneDehumidNum > NumDehumidifiers || ZoneDehumidNum < 1) {
                 ShowFatalError(state,
                                format("SimZoneDehumidifier:  Invalid CompIndex passed= {}, Number of Units= {}, Entered Unit name= {}",
                                       ZoneDehumidNum,
-                                      state.dataZoneDehumidifier->NumDehumidifiers,
+                                      NumDehumidifiers,
                                       CompName));
             }
-            if (state.dataZoneDehumidifier->CheckEquipName(ZoneDehumidNum)) {
+            if (state.dataZoneDehumidifier->ZoneDehumid(ZoneDehumidNum).CheckEquipName) {
```

Comment:
Would help to use `auto & zoneDehumid = state.dataZoneDehumidifier->ZoneDehumid(ZoneDehumidNum);` here.